src/IO/WriteBufferFromS3.h

```diff
@@ -88,7 +88,9 @@ class WriteBufferFromS3 final : public BufferWithOwnMemory<WriteBuffer>
 
     const String bucket;
     const String key;
-    const S3Settings::RequestSettings request_settings;
+    const S3Settings::RequestSettings::PartUploadSettings settings;
+    const bool check_objects_after_upload = false;
+    const size_t max_unexpected_write_error_retries = 4;
```

Comment:
(Sounds like this should be made configurable as setting instead of a hardcoded constant.)

Reply:
It is: `s3_max_unexpected_write_error_retries`